Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -98,59 +98,29 @@ open class MapView: UIView {
      */
     @available(iOS, deprecated: 1000000)
     public var preferredFramesPerSecond: Int {
-        get {
-            return _preferredFramesPerSecond ?? 0
-        }
-        set {
-            _preferredFramesPerSecond = newValue
-        }
-    }
-
-    private var _preferredFramesPerSecond: Int? {
-        didSet {
-            updateDisplayLinkPreferredFramesPerSecond()
-        }
+        get { displayLink.preferredFramesPerSecond }
+        set { displayLink.preferredFramesPerSecond = newValue }
     }
 
     // Checking Swift version as a proxy for iOS SDK version to enable
     // building with iOS SDKs < 15
-    #if swift(>=5.5)
+#if swift(>=5.5)
     /// The preferred range of frame refresh rates.
     @available(iOS 15.0, *)
     public var preferredFrameRateRange: CAFrameRateRange {
-        get {
-            return _preferredFrameRateRange ?? .default
-        }
-        set {
-            _preferredFrameRateRange = newValue
-        }
+        get { displayLink.preferredFrameRateRange }
+        set { displayLink.preferredFrameRateRange = newValue }
     }
-
-    // Stored properties cannot be annotated with @available, so we
-    // store the value as an `Any` in `_untypedPreferredFrameRateRange` below
-    // and make this a computed property.
-    @available(iOS 15.0, *)
-    private var _preferredFrameRateRange: CAFrameRateRange? {
-        get {
-            return _untypedPreferredFrameRateRange as? CAFrameRateRange
-        }
-        set {
-            _untypedPreferredFrameRateRange = newValue
-            updateDisplayLinkPreferredFramesPerSecond()
-        }
-    }
-
-    private var _untypedPreferredFrameRateRange: Any?
-    #endif
+#endif
 
     /// The `timestamp` from the underlying `CADisplayLink` if it exists, otherwise `nil`
     @_spi(Metrics) public var displayLinkTimestamp: CFTimeInterval? {
-        return displayLink?.timestamp
+        return displayLink.timestamp.isNaN ? nil : displayLink.timestamp
     }
 
     /// The `duration` from the underlying `CADisplayLink` if it exists, otherwise `nil`
     @_spi(Metrics) public var displayLinkDuration: CFTimeInterval? {
-        return displayLink?.duration
+        return displayLink.duration.isNaN ? nil : displayLink.duration
     }
```

Comment:
We should add some simple unit tests for these.

Reply:
I think `testDisplayLinkTimestampIsNilWhenDisplayLinkIsEmpty`, `testDisplayLinkTimestampWhenDisplayLinkIsNotEmpty`, `testDisplayLinkDurationIsNilWhenDisplayLinkIsEmpty` and `testDisplayLinkDurationWhenDisplayLinkIsNotEmpty` should cover these.